### DISCUSSION TITLE
Non linear motor rotation

### DISCUSSION THREAD

**Discussion by pirsquared** [pybricks/support#1845] [Q&A] [answered]
Hi Community,

My team is attempting to run a motor driving a forklift and we want the lift to go up and then down in one smooth motion.  We can can use motor.run_angle(x) followed by motor.run_angle(-x) but this is not very smooth.  And, what's more, is that it seems as if there is a pause between the processing of one command and the processing of the subsequent command.  Ideally, I like to see the motor run back then forth in a sinusoidal manner.

Thanks for any feedback!

**Comment by laurensvalk** [comment, marked as answer]
How fast is your motor turning? With slow speeds and high load, it may not be very smooth, in which case a sinusoidal motion may not necessarily resolve it. If this is the case, gearing down the mechanism usually helps.

> it seems as if there is a pause between the processing of one command and the processing of the subsequent command.

Processing is fast, but the motors will try to wait until they have reached their target before embarking on the next move. If it can't quite reach it due to limited torque or load, it will appear to be slow.

I would also generally recommend using `run_target` for mechanisms like these, so you can make it run to the start and end with fixed values.

If you do want to make sinusoidal motion, you could use `track_target` repeatedly, each time giving a new angle along the sinusoidal curve.

**Reply by pirsquared** [reply]
Thanks, we’ll give that a try and I’ll reply back.

**Reply by pirsquared** [reply]
This worked out perfectly.  I just ran a while loop and each loop I calculated where I should be based on the elapsed time and targeted that with `run_target`.

Thanks again.
